packages/python/plotly/plotly/io/_utils.py

```diff
@@ -1,6 +1,29 @@
 import plotly
 import plotly.graph_objs as go
 from plotly.offline import get_plotlyjs_version
+from plotly import optional_imports
+
+import warnings
+import psutil
+
+jupyter_notebook = optional_imports.get_module("notebook")
+jupyter_lab = optional_imports.get_module("jupyterlab")
+
+
+def display_jupyter_version_warnings():
+
+    parent_process = psutil.Process().parent().cmdline()[-1]
```

Comment:
IDK how this would happen in practice, but in principle `.parent()` can [return None](https://psutil.readthedocs.io/en/latest/#psutil.Process.parent) which would make this fail. In that case I guess we aren't in a jupyter context so we can short-circuit the rest of this.

Reply:
I added some code to address this! Let me know if you think there's a better approach than a try/except